src/CodeGen/ODataT4CodeGenerator.cs

```diff
@@ -425,6 +435,123 @@ public void ValidateAndSetIgnoreUnexpectedElementsAndAttributesFromString(string
     this.IgnoreUnexpectedElementsAndAttributes = boolValue;
 }
 
+public void ValidateAndSetCustomHttpHeadersFromString(string stringValue) 
+{
+    this.CustomHttpHeaders = new Dictionary <string, string> ();
+
+    if (String.IsNullOrWhiteSpace(stringValue)) {
+        return;
+    }
+
+    if (ValidateParantheses(stringValue) && ValidateFirstLast(stringValue) && ValidateDelimeters(stringValue)) 
+    {
+        string header = stringValue.TrimEnd('}', ' ');
+        string[] elements = header.Split('}');
+        foreach (var element in elements)
+        {
+            string _element = element.TrimStart(',', '{', ' ');
+            string[] array = _element.Split(':');
+            CustomHttpHeaders.Add(array[0], array[1]);
+        }
+    }
+    else 
+    {
+        throw new ArgumentException("Please check your custom http headers string format! You have set the custom http header but not use the format '{headerName:value},{headerName:value}'");
+    }
+
+}
+
+private static bool ValidateParantheses(string _input)
+{
+    Dictionary<char, char> paraPairs = new Dictionary<char, char>() {
+        { '{', '}' }
+    };
+
+    Stack<char> stack = new Stack<char>();
+
+    foreach (char i in _input)

```

Comment:
"i" looks an integer. maybe use "c" or "character" is better?

Reply:
I have made the necessary changes